docs/proposals/implement-api-that-imports-prometheus-blocks-into-cortex.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Implement API that imports Prometheus blocks into Cortex"
+linkTitle:
+weight:
+slug:
+---
+
+- Author: [Siddharth Asthana](https://github.com/edith007)
+- Date: March 2023
+- Status: draft
+
+## Problem statement
+
+There might be a lot of cases where the users might already be using Prometheus and later decides to switch to Cortex. Cortex will take care of the metrics which will be generated after the switch, but at present there is no way to backfill the old Prometheus TSDB blocks to Cortex. This proposal is to add the ability in cortex to fetch the Prometheus TSDB blocks and backfill the data.
+
+## Detailed Solution
+
+We can create a new service called `Importer` which will be responsible for bringing in the old Prometheus TSDB blocks to Cortex. 
+
+### How Importer will work
+
+The importer service will have two API endpoints: `import` and `import/status/{id}`
+
+`import`:
+
+- It will take a link to where the old Prometheus TSDB blocks are stored
+- The API will return an ID. This ID can be used to get the status of the import operation
+- Cortex will download this block from the link received and upload to the long term storage
+
+`import/status/{id}`:
+
+- This endpoint can be used to check the status of a particular import operation.
+- It will return status as `SUCCESS`, `FAILED` or `ONGOING`.
```

Comment:
Will we cover more information about the current status? For example, when you upload something it is handy to see a progress bar and % of data uploaded.

Reply:
Also we should consider partial failures.

Reply:
Thanks for the input @yeya24 🙇🏻  I Will add that too!